### PR TITLE
Conditional requirements 1, 3, and 4 for CSV formats

### PR DESCRIPTION
**Pull request by mint-thompson**
Add conditional check for payer specific charge information.
Add conditional check for additional notes when methodology is "other".
Add conditional check for at least one standard charge value being available for an item or service.

Please take your time reading and enjoying the new tests. In particular, please check that I've covered cases where a number/enum field has an invalid value and is not required.

### CODE REVIEW

**Review by carrils — APPROVED** [review]
Very good work Mint! Thank you

File: test/2.0/csv.spec.ts [hunk 1]
```diff
@@ -395,6 +395,240 @@ test("validateRow tall conditionals", (t) => {
...
+  t.is(otherWithoutNotesErrors.length, 1)
+  t.is(
+    otherWithoutNotesErrors[0].message,
+    'If the "standard charge methodology" encoded value is "other", there must be a corresponding explanation found in the "additional notes" for the associated payer-specific negotiated charge.'
+  )
+
+  // If an item or service is encoded, a corresponding valid value must be encoded for at least one of the following:
+  // "Gross Charge", "Discounted Cash Price", "Payer-Specific Negotiated Charge: Dollar Amount", "Payer-Specific Negotiated Charge: Percentage",
+  // "Payer-Specific Negotiated Charge: Algorithm".
+  const itemNoChargeRow = {
```
Comment (shaselton-usds) [inline test/2.0/csv.spec.ts:558]: Just documenting for observed behavior:

If one of the standard charges is populated with an illegal value and the others left blank, an error is thrown about the illegal value and the `ITEM_REQUIRES_CHARGE` isn't thrown.

I think this is fine.

File: test/2.0/csv.spec.ts [hunk 2]
```diff
@@ -722,11 +956,313 @@ test("validateRow wide conditionals", (t) => {
...
+  const dollarWithMethodologyErrors = validateRow(
+    dollarWithMethodologyRow,
+    5,
+    columns,
+    true
+  )
+  t.is(dollarWithMethodologyErrors.length, 0)
+  const dollarNoMethodologyRow = {
+    ...basicRow,
+    "standard_charge | Payer One | Basic Plan | negotiated_dollar": "552",
```
Comment (shaselton-usds) [inline test/2.0/csv.spec.ts:977]: I ran this test with the follow parameters that generated no errors:

```json 
const dollarNoMethodologyRow = {
     ...basicRow,
     "standard_charge | Payer One | Basic Plan | negotiated_algorithm": "",
     "standard_charge | Payer One | Basic Plan | negotiated_dollar": "",
     "standard_charge | Payer One | Basic Plan | negotiated_percentage": "",
     "standard_charge | Payer One | Basic Plan | methodology": "fee schedule",
   }
```
Should this trigger the `ITEM_REQUIRES_CHARGE` error?

File: test/2.0/csv.spec.ts [hunk 3]
```diff
@@ -722,11 +956,313 @@ test("validateRow wide conditionals", (t) => {
...
+  t.assert(
+    percentageNoMethodologyErrors[0].message.includes(
+      '"standard_charge | Payer Two | Special Plan | methodology" is required when a payer specific negotiated charge is encoded as a dollar amount, percentage, or algorithm'
+    )
+  )
+  const percentageWrongMethodologyRow = {
+    ...basicRow,
+    "standard_charge | Payer Two | Special Plan | negotiated_percentage": "50",
+    "estimated_amount | Payer Two | Special Plan": "244",
+    "standard_charge | Payer One | Basic Plan | methodology": "fee schedule",
```
Comment (shaselton-usds) [inline test/2.0/csv.spec.ts:1042]: This can exist without a standard charge?

File: test/2.0/csv.spec.ts [hunk 4]
```diff
@@ -722,11 +956,313 @@ test("validateRow wide conditionals", (t) => {
...
+  )
+
+  // If the "standard charge methodology" encoded value is "other", there must be a corresponding explanation found
+  // in the "additional notes" for the associated payer-specific negotiated charge.
+  const otherWithNotesRow = {
+    ...basicRow,
+    "standard_charge | Payer One | Basic Plan | negotiated_percentage": "80",
+    "estimated_amount | Payer One | Basic Plan": "150",
+    "standard_charge | Payer One | Basic Plan | methodology": "other",
+    "additional_payer_notes | Payer One | Basic Plan":
```
Comment (shaselton-usds) [inline test/2.0/csv.spec.ts:1135]: This wouldn't satisfy the "additional notes" being populated for the specific plan's "other" in methodology?

File: test/2.0/csv.spec.ts [hunk 5]
```diff
@@ -722,11 +956,313 @@ test("validateRow wide conditionals", (t) => {
...
+  t.is(otherWrongNotesErrors.length, 1)
+  t.is(
+    otherWrongNotesErrors[0].message,
+    'If the "standard charge methodology" encoded value is "other", there must be a corresponding explanation found in the "additional notes" for the associated payer-specific negotiated charge.'
+  )
+
+  // If an item or service is encoded, a corresponding valid value must be encoded for at least one of the following:
+  // "Gross Charge", "Discounted Cash Price", "Payer-Specific Negotiated Charge: Dollar Amount", "Payer-Specific Negotiated Charge: Percentage",
+  // "Payer-Specific Negotiated Charge: Algorithm".
+  const itemNoChargeRow = {
```
Comment (shaselton-usds) [inline test/2.0/csv.spec.ts:1180]: Duplicate test found on line 558?

**Review by shaselton-usds — APPROVED** [review]
Approving PR with outstanding comments to potentially be addressed in a future PR.

File: src/versions/2.0/csv.ts [hunk 6]
```diff
@@ -551,48 +554,140 @@ function validateModifierRow(
...
+        csvErr(
+          index,
+          columns.indexOf(chargeColumns[0]),
+          chargeColumns[0],
+          ERRORS.ITEM_REQUIRES_CHARGE()
+        )
+      )
+    }
+  }
+  // payersPlans.forEach(([payer, plan]) => {
```
Comment (shaselton-usds) [inline src/versions/2.0/csv.ts:670]: We can probably remove this, yes?

**Pull request merged**